Model/Export/SourceIterator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Hyva\Admin\Model\Export;
+
+use Hyva\Admin\ViewModel\HyvaGrid\NavigationInterface;
+use Hyva\Admin\ViewModel\HyvaGrid\RowInterface;
+use Hyva\Admin\ViewModel\HyvaGridInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+
+class SourceIterator implements \Iterator
+{
+    protected SearchCriteriaInterface $searchCriteria;
+    protected int $total;
+    protected NavigationInterface $navigation;
+
+    private HyvaGridInterface $grid;
+
+    private array $currentBatch = [];
+    private int $currentCounter=0;
+
+    public function __construct(HyvaGridInterface $grid)
+    {
+        $this->grid = $grid;
+        $this->navigation = $grid->getNavigation();
+        $this->searchCriteria = $this->navigation->getSearchCriteria();
+        $this->searchCriteria->setPageSize(200);
+        $this->total = $this->navigation->getTotalRowsCount();
+    }
+
+    public function current(): RowInterface
+    {
+        if(!isset($this->currentBatch[$this->currentCounter])){
+            $this->currentBatch = [];
+            $page =  (int) ceil($this->currentCounter / $this->searchCriteria->getPageSize());
+            $this->searchCriteria->setCurrentPage($page + 1);
+            $inBatchCounter = 0;
+            foreach( $this->grid->getRows() as $row){
```

Comment:
Wouldn't it then be easer or more readable to pass the serachCriteria to th getRow method in the HyvaGridViewModel?
Because inside of that method the searchCriteria is fetched from the navigatoin.

Or how about creating a new method in that class that builds the rows with a search criteria?